pkg/connector/slack/v0/apiFunctions.go

```diff
@@ -0,0 +1,184 @@
+package slack
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/slack-go/slack"
+)
+
+func loopChannelListAPI(e *execution, isPublic bool, channelName string, targetChannelID *string) error {
+	var apiParams slack.GetConversationsParameters
+	setChannelType(&apiParams, isPublic)
+
+	for {
+
+		slackChannels, nextCur, err := e.client.GetConversations(&apiParams)
+		if err != nil {
+			return err
+		}
+
+		setChannelID(channelName, slackChannels, targetChannelID)
+
+		if *targetChannelID != "" {
+			break
+		}
+
+		if *targetChannelID == "" && nextCur == "" {
+			err := fmt.Errorf("there is no match name in slack channel [%v]", channelName)
+			return err
+		}
+
+		apiParams.Cursor = nextCur
+
+	}
+
+	return nil
+}
+
+// Todo: make it multiple options
+func setChannelType(params *slack.GetConversationsParameters, isPublicChannel bool) {
+	if !isPublicChannel {
+		params.Types = append(params.Types, "private_channel")
+	} else {
+		params.Types = append(params.Types, "public_channel")
+	}
+}
+
+func setChannelID(channelName string, channels []slack.Channel, targetChannelID *string) {
```

Comment:
`func getChannelID(channelName string, channels []slack.Channel, targetChannelName string) (channelID string)`
I will not return error because it could be match in the next fetch of API.